src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeDeclarationCheck.java

```diff
@@ -0,0 +1,122 @@
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.ScopeUtil;
+
+/**
+ * <p>
+ * Checks if unnecessary semicolon is used after type declaration.
+ * </p>
+ * <p>
+ * This check is not applicable to nested type declarations,
+ * <a
+ * href="https://checkstyle.org/config_coding.html#UnnecessarySemicolonAfterTypeMemberDeclaration">
+ * UnnecessarySemicolonAfterTypeMemberDeclaration</a> is responsible for it.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code tokens} - tokens to check
+ * Default value is:
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CLASS_DEF">
+ * CLASS_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INTERFACE_DEF">
+ * INTERFACE_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF">
+ * ENUM_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_DEF">
+ * ANNOTATION_DEF</a>.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;UnnecessarySemicolonAfterTypeDeclaration&quot;/&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class A {
+ *
+ *     class Nested {
+ *
+ *     }; // OK, nested type declarations are ignored
+ *
+ * }; // violation
+ *
+ * interface B {
+ *
+ * }; // violation
+ *
+ * enum C {
+ *
+ * }; // violation
+ *
+ * {@literal @}interface D {
+ *
+ * }; // violation
+ * </pre>
+ *
+ * @since 8.31
+ */
+@StatelessCheck
+public final class UnnecessarySemicolonAfterTypeDeclarationCheck extends AbstractCheck {
```

Comment:
Should this be `UnnecessarySemicolonAfter`Outer`TypeDeclarationCheck`.

Reply:
Good question. 
@romani what do you think, should "Outer" be added to check name?

Reply:
@strkkk , yes, please change a name. To be more exact by name.
sorry I did not noticed this unresolved conversation before.

Reply:
@romani name has been changed already, I also updated original issue and contribution repo PR.
Only diff report has previous name, but it is ok

Reply:
confirmed.